Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -284,8 +282,10 @@ static void drawCellFocusRingInView(const FloatRect& rect, NSCell *cell, NSView
     [cell drawFocusRingMaskWithFrame:rect inView:view];
 }
 
-void ControlMac::drawCellFocusRingInternal(const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
+void ControlMac::drawCellFocusRingInternal(GraphicsContext& context, const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style, NSCell *cell)
 {
+    LocalCurrentGraphicsContext localContext(context);
```

Comment:
I'm assuming focus rings don't suffer from the flipping issue when used without a view?

Reply:
We can't tell until rdar://105249508 is fixed. The focus ring is not displayed with viewless drawing. But also we can't create an NSView out of the main thread. This is why the focus ring is not working with GPUP for DOM rendering is on.